account_move_locking/models/date_range.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+# © 2016 Camptocamp SA (Matthieu Dietrich)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+from openerp import _, api, exceptions, fields, models
+
+
+class DateRange(models.Model):
+    _inherit = "date.range"
+
+    @api.depends('locked_journal_ids')
+    def _get_lock_state(self):
+        journal_obj = self.env['account.journal']
+        for range in self:
+            all_journals = journal_obj.search(
+                [('company_id', '=', range.company_id.id)])
+            if not range.locked_journal_ids:
+                range.lock_state = 'unlocked'
+            elif range.locked_journal_ids == all_journals:
+                range.lock_state = 'locked'
+            else:
+                range.lock_state = 'partial'
+
+    lock_state = fields.Selection(
+        [('unlocked', 'Unlocked'),
+         ('partial', 'Partially locked'),
+         ('locked', 'Locked')],
+        compute="_get_lock_state",
+        readonly=True,
+        store=True)
+    locked_journal_ids = fields.Many2many(
+        'account.journal',
+        rel='account_date_range_lock_journal',
+        string="Locked journals")
+
+    @api.model
+    def create(self, vals):
+        # Check if fiscal year is locked; if so, add all journals
+        # to set is as locked
```

Comment:
The lock can be on 2 levels:
- the fiscal year is the "master" lock : once it is locked, it cannot be unlocked.
- below that, you can have other date ranges, monthly periods for example ; those are "children", and can be locked and unlocked as long as the fiscal year "above" those are not locked.

The goal in this particular method is that, if you create another period in an already-locked fiscal year, it should be locked as well. It's not necessary on an account entry level (as the FY will forbid any modification), but it's clearer to the user that every period inside a locked FY is locked as well.